### ISSUE TITLE
BatchNorm momentum set to 0 in Inception models

### ISSUE THREAD

**Issue by rwightman** [Cadene/pretrained-models.pytorch#5]
Both Inception models initialize BatchNorm with self.bn = nn.BatchNorm2d(out_planes, eps=0.001, momentum=0, affine=True)

Setting momentum to 0 prevents the running mean and var from being updated during training. When training one of these models on a different dataset, this causes a significant discrepancy between train and eval modes.

**Comment by Cadene** [comment]
Thanks for submitting this issue. However, I am not fluent in tensorflow. What should be the value of the momentum then?

The [batch_norm in resnetinceptionv2 of tensorflow](https://github.com/Cadene/tensorflow-model-zoo.torch/blob/master/models/slim/nets/inception_resnet_v2.py#L254) has 3 parameters:

- weight_decay
- batch_norm_decay
- batch_norm_epsilon

The [batch_norm of pytorch](http://pytorch.org/docs/master/nn.html#torch.nn.BatchNorm2d) has 2 parameters:

- eps
- momentum

Most probably, setting the momentum to its default pytorch value (0.1) should be enough. Did you try something?

**Comment by rwightman** [comment]
I removed all the extra parameters and used the Pytorch defaults. Worked
fine for my purposes and in the majority of other image classification
networks I've seen built in Pytorch.

**Comment by Cadene** [comment]
Setting the momentum to a non zero value, doesn't affect the validation process as expected. But changing the epsilon value does. However it has a minor impact.

inceptionv4:

- eps=0.001 ; Acc@1 80.062 ; Acc@5 94.924
- eps=1e-5 ; Acc@1 80.54 ; Acc@5 94.926

I should probably keep `eps` to 0.001 and set `momentum` to 0.1. However, I am open to suggestions.

**Comment by rwightman** [comment]
The epsilon is used for numerical stability calculating the variance so it's impact will be minimal. The momentum value won't be noticed until you continue to train the model with data that has a different mean/variance from the data that the pretrained model was trained with (Imagenet), you will then notice, after switching to eval after some time training, that your network performance is awful compared to performance while training where the current batch mean/var is being used exclusively and the running mean/var were not being updated.
